Frontend/banner

YuanshuoDu opened the pull request:
### Description

Add banner

Fixes # (issue)

### Type of change

Please select the option that best describes the changes made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Changes

Added banner and optimise the position

1Solon commented:
@Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

1Solon reviewed (changes requested):
Please fix the linting issues

1Solon commented on Frontend/src/app/terms/page.tsx:
This should be split this off into a new PR?

YuanshuoDu replied:
Have splitted into terms pr

1Solon commented on Frontend/src/app/signup/page.tsx:
This should be split off into the same PR as terms

YuanshuoDu replied:
has merged the terms into the banner

YuanshuoDu replied:
Have splitted

YuanshuoDu commented:
> Hey @YuanshuoDu we have moved away from the cookie based auth while your code here checks for `magpie_auth` cookie. Can we change that please?
> 
> ```ts
> useEffect(() => {
>     if (isPageLoaded) {
>       // get cookie
>       const cookieString = document.cookie;
>       const cookies = cookieString.split('; ').reduce((acc, cookie) => {
>         const [name, value] = cookie.split('=');
>         acc[name] = value;
>         return acc;
>       }, {} as Record<string, string>);
> 
>       // check if cookie exists or expired
>       const hasCookie = cookies['magpie_auth'];
>       const expiryTime = cookies['magpie_auth_expiry']; 
> 
>       if (hasCookie && expiryTime) {
>         const expiryDate = new Date(parseInt(expiryTime, 10));
>         const now = new Date();
>         if (now < expiryDate) {
>           setShow(false);
>         } else {
>           setShow(true);
>         }
>       } else {
>         setShow(true);
>       }
>     }
>   }, [isPageLoaded])
> ```

has called new functions instead of old cookies

YuanshuoDu commented:
> @Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

![image](https://github.com/user-attachments/assets/4664abc5-79b9-407b-a256-b57b02516419)
the alert dialog need the trigger and it also will set the background into blur

YuanshuoDu commented:
> @Stevendu666 why use an [alert](https://ui.shadcn.com/docs/components/alert) rather then an [alert dialogue?](https://ui.shadcn.com/docs/components/alert-dialog)

![image](https://github.com/user-attachments/assets/431c23e1-dac1-4146-8a0e-9766224e248d)
has implemented new UI instead of using alert

1Solon commented on Frontend/src/app/components/banner/CookieConsent.tsx:
```diff
@@ -0,0 +1,153 @@
+  onDeclineCallback = async () => {
+    try {
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
```
It's not necessary to log this to console

YuanshuoDu replied:
Deleted

1Solon commented on Frontend/src/app/components/banner/CookieConsent.tsx:
```diff
@@ -0,0 +1,153 @@
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
+    } catch (error) {
+      console.error("Error during handling deny:", error);
```
Here also

YuanshuoDu replied:
Deleted as well

1Solon commented on Frontend/src/app/components/banner/CookieConsent.tsx:
```diff
@@ -0,0 +1,153 @@
+  const accept = async () => {
+    setIsOpen(false);
+    document.cookie =
+      "cookieConsent=true; expires=Fri, 31 Dec 9999 23:59:59 GMT";
```
We have an existing cookie to track whether the cookie was accepted- this is triggered by the `setCookiesAccepted();` you called earlier.  

Additionally, hard-coding an expiry value like this potentially opens us to issues. Please coordinate with the team if you plan on editing cookie logic in future like this.

YuanshuoDu replied:
Deleted hard-coding, called setCookiesAccepted, commitSessionToCookies

1Solon commented on Frontend/src/app/components/banner/CookieConsent.tsx:
```diff
@@ -0,0 +1,153 @@
+  useEffect(() => {
+    try {
+      setIsOpen(true);
+      if (document.cookie.includes("cookieConsent=true")) {
```
This should likely be `getCookiesAccepted`, we have predefined behaviour to check if the cookie has already been done. Wherever possible, we should aim not to duplicate code.

YuanshuoDu replied:
replaced to getCookiesAccepted

1Solon reviewed (changes requested):
![image](https://github.com/user-attachments/assets/46fc6ac0-2a2f-458e-b089-2cb5fec0537d)

The cookie banner should probably be further right then it is currently, as it conflicts with the onboarding button

YuanshuoDu commented:
> ![image](https://private-user-images.githubusercontent.com/55561235/386398852-46fc6ac0-2a2f-458e-b089-2cb5fec0537d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MzI4ODMsIm5iZiI6MTczMTYzMjU4MywicGF0aCI6Ii81NTU2MTIzNS8zODYzOTg4NTItNDZmYzZhYzAtMmEyZi00NThlLWIwODktMmNiNWZlYzA1MzdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE1VDAxMDMwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5MGU1NDdmZGIxNWE2YjBlM2MzOWI5NzliYmQ0OGVmYTQ3YmVhOTE2NWRlMjc5NGUwYTBmNDcyNGIzMzVjYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9z2RtH2fTr2Dmfic7tMbtjXUWYIpdD1Eb2FMT8XVIL8)
> 
> The cookie banner should probably be further right then it is currently, as it conflicts with the onboarding button

using the percentage value to left instead of set value to avoid different screen showing different
